python/ray/_private/log_monitor.py

```diff
@@ -440,6 +447,14 @@ def run(self):
                 time.sleep(0.1)
 
 
+def is_proc_alive(pid):
+    try:
+        return psutil.Process(pid).is_running()
+    except Exception:
+        # If OSError is raised, the process does not exist.
+        return False
+
+
```

Comment:
Moved the function to global space so it could be used in testing, and changed the implementation to use psutil, which was the crux of what I wanted to do in this PR